cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -212,6 +213,16 @@ public void updateSchedule(final ProgramSchedule schedule) throws NotFoundExcept
       throw new IllegalStateException(
         "Schedule '" + schedule.getScheduleId() + "' already exists despite just being deleted.");
     }
+    // if the schedule was previously enabled, it should still/again enabled be after the update
+    if (ProgramScheduleStatus.SCHEDULED == previousStatus) {
+      try {
+        enableSchedule(schedule.getScheduleId());
```

Comment:
during app deployment, the `addSchedule` is a separate call from `enableSchedule`?
Intuitively, it seems like a schedule should be enabled within the `addSchedule`